### PR TITLE
Add `under normal circumstances` to `cell` docs

### PR DESCRIPTION
**Pull request by CJendantix**
Someone commented that it was a bit misleading that the docs of `cell` said that you could never get a reference to the interior value of a `Cell`, when you can get a mutable reference to the underlying data if `self` is mutable. Shouldn't be a controversial addition.

### CODE REVIEW

**Comment by workingjubilee** [comment]
Is "holding `&mut Cell<T>`" really *not* "normal circumstances"?

File: library/core/src/cell.rs [hunk 1]
```diff
@@ -28,10 +28,10 @@
...
 //!
 //! [`Cell<T>`] implements interior mutability by moving values in and out of the cell. That is, an
-//! `&mut T` to the inner value can never be obtained, and the value itself cannot be directly
-//! obtained without replacing it with something else. Both of these rules ensure that there is
-//! never more than one reference pointing to the inner value. This type provides the following
-//! methods:
+//! `&mut T` to the inner value can never be obtained under normal circumstances, 
+//! and the value itself cannot be directly obtained without replacing it with something else. 
+//! Both of these rules ensure that there is never more than one reference pointing to the inner value.
+//! This type provides the followin methods:
```
Comment (workingjubilee) [inline library/core/src/cell.rs:34]:
```suggestion
//! This type provides the following methods:
```

**Comment by workingjubilee** [comment]
Hm. I will leave that decision to jhpratt! It just made me wonder.

**Comment by jhpratt** [comment]
`Cell::get_mut` _does_ exist, so the docs aren't 100% correct. With that said, I wouldn't consider having ownership of a `Cell` to be out of the ordinary (passing a mutable reference would be odd, though).

I'm open to some change in wording here; my preferred solution would be to differentiate obtaining a mutable reference at runtime (as with `RefCell`) versus requiring `&mut self` to prove exclusivity at compile time.

**Comment by Dylan-DPC** [comment]
@CJendantix any updates on this? thanks

**Comment by SkyeLumine** [comment]
This documentation threw me for a loop when I read it. The language, _"an &mut T to the inner value can never be obtained"_ is just not true when taken at face value (`Cell<T>` clearly has `pub fn get_mut(&mut self) -> &mut T`). I think _"under normal circumstances"_ is an improvement, but it could definitely still be more clear. I think the main case we're considering here is that `&mut T` can't be obtained through a shared reference. So, how about this? _"That is, an &mut T to the inner value can never be obtained **from a shared reference**, and the value itself cannot be directly obtained without replacing it with something else."_

**Pull request closed without merging**